tests/test_padding.py

```diff
@@ -25,19 +26,35 @@
 
 
 # Helper function to create a 3D array and remap it to the global array
-def create_spmd_array(global_shape, pdims):
+def create_spmd_array(global_shape, pdims, complex=False):
 
   assert (len(global_shape) == 3)
   assert (len(pdims) == 2)
   assert (prod(pdims) == size
          ), "The product of pdims must be equal to the number of MPI processes"
 
-  local_array = jax.random.normal(
-      shape=[
-          global_shape[0] // pdims[1], global_shape[1] // pdims[0],
-          global_shape[2]
-      ],
-      key=jax.random.PRNGKey(rank))
+  if complex:
+
+    local_array = jax.random.normal(
+        shape=[
+            global_shape[0] // pdims[1], global_shape[1] // pdims[0],
+            global_shape[2]
+        ],
+        key=jax.random.PRNGKey(rank)) + 1j * jax.random.normal(
+            shape=[
+                global_shape[0] // pdims[1], global_shape[1] // pdims[0],
+                global_shape[2]
+            ],
+            key=jax.random.PRNGKey(rank + 1))
```

Comment:
I know this is just for tests, but I imagine that since the rank is the process index, this offset should be bigger than this, or you could just split the previous key.